thriftserver/server/src/main/scala/org/apache/livy/thriftserver/cli/ThriftCLIService.scala

```
@@ -83,6 +84,22 @@ abstract class ThriftCLIService(val cliService: LivyCLIService, val serviceName:
     super.init(livyConf)
   }
 
+  protected def getKeyStorePassword(): String = {
+    val credentialProviderPath = livyConf.get(LivyConf.HADOOP_CREDENTIAL_PROVIDER_PATH)
+    val hadoopConf = new Configuration()
+    if (credentialProviderPath != null) {
+      hadoopConf.set("hadoop.security.credential.provider.path", credentialProviderPath)
+    }
+    Option(livyConf.get(LivyConf.SSL_KEYSTORE_PASSWORD))
+      .orElse {
+        Option(hadoopConf.getPassword(LivyConf.SSL_KEYSTORE_PASSWORD.key)).map(_.mkString)
+      }
+      .getOrElse {
+        throw new IllegalArgumentException(
```

why don't we return just null as before here?

---

Before I made the changes here, I found that if the keystore password was not set in the server's livy.conf, as it defaults to null, the null causes an NPE in the Thrift code that gets called subsequently, specifically in https://github.com/apache/thrift/blob/0.9.3/lib/java/src/org/apache/thrift/transport/TSSLTransportFactory.java#L199. I thought it would be better to throw an exception here up front specifying the problem. This is also consistent with throwing an IllegalArgumentException when the keystore path is not configured.